SeaPublicWebsite/wwwroot/js/bundle/google-analytics.js

```diff
@@ -0,0 +1,139 @@
+﻿// Send a hit when text is copied or pasted
+function selectAndCopy() {
+    document.addEventListener("copy", () => {
+        gtag("event", "copy_text", {
+            "value": document.getSelection().toString()
+        });
+    });
+    document.addEventListener("paste", (e) => {
+        gtag("event", "paste_text", {
+            "value": e.clipboardData.getData("text")
+        });
+    });
+}
+
+// For checkboxes: Send a hit when a checkbox becomes unchecked
+// For radios and text inputs: Send a hit when the value of the input is changed (excluding when no value exists or none was selected)
+function changeInputValue() {
+    const inputs = document.getElementsByTagName("input");
+    const inputAnswers = new Map();
+
+    for (const input of inputs) {
+        input.addEventListener("change", () => {
+            let name = input.name;
+            let value = input.value;
+            let type = input.type;
+            
+            if (type === "checkbox") {
+                if (!input.checked) {
+                    gtag("event", `change_input_${type}`, {
+                        "name": name,
+                        "value": value,
+                    });
+                }
+            } else if (type === "radio" || type === "text") {
```

Comment:
For text inputs does this event fire only when the control loses focus, or every time the user types a character?

Reply:
Yes, that is why I used onChange. The event is only fired when focus is moved away.